plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableQueryTests.php

```diff
@@ -235,4 +236,43 @@ public function test_query_filters() {
 		remove_all_filters( 'woocommerce_orders_table_query_clauses' );
 	}
 
+	/**
+	 * @testdox The pre-query escape hook allows replacing the order query.
+	 */
+	public function test_pre_query_escape_hook() {
+		$order1 = new \WC_Order();
+		$order1->set_date_created( time() - HOUR_IN_SECONDS );
+		$order1->save();
+
+		$order2 = new \WC_Order();
+		$order2->save();
+
+		$query = new OrdersTableQuery( array() );
+		$this->assertCount( 2, $query->orders );
+		$this->assertEquals( 2, $query->found_orders );
+		$this->assertEquals( 0, $query->max_num_pages );
+
+		$callback = function( $query ) use ( $order1 ) {
+			// Only return one of the orders to show that we are replacing the query result.
+			$order_ids = array( $order1->get_id() );
+			// These are made up to show that we are actually replacing the values. 
+			$found_orders = 17;
+			$max_num_pages = 23;
+			return [ $order_ids, $found_orders, $max_num_pages ];
+		};
+		add_filter( 'woocommerce_hpos_pre_query', $callback, 10, 3 );
```

Comment:
The filter only has 2 parameters and the callback in the test only uses the first one. The `, 10, 3` can be removed.